rfcs/0028-command-line-registry-composition.md

```diff
@@ -0,0 +1,153 @@
+- Feature Name: Command Line Composition from Internal Registry
+- Start Date: 2021-08-24
+- RFC PR: [apache/tvm-rfcs#28](https://github.com/apache/tvm-rfcs/pull/28)
+- GitHub Issue: [apache/tvm#0000](https://github.com/apache/tvm/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Introducing a standardised form for `tvmc` arguments to be populated from internal registries in TVM.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, when a user uses `tvmc`, they present a target string:
+```
+tvmc --target="woofles -mcpu=woof, c -mattr=+mwoof"
+```
+
+Using a target string here means that the entire `--target` argument is used as an opaque pass-through to the internal `Target` string parser. Users coming to TVM should be able to compose these options and get meaningful help when doing that composition with `tvmc`. As an example, using the default `argparse` behaviour, the arguments can be registered as follows:
```

Comment:
let us motivate composition of Target a bit further. what use case are we trying to support? what user burden are we trying to alleviate?

Reply:
move the argparse code example to Guide-level explanation. i think the example `--help` output could stay as motivation.

Reply:
> let us motivate composition of Target a bit further. what use case are we trying to support? what user burden are we trying to alleviate?

I've tried to clarify this with some additional text, I've also added a series of disclaimers in the RFC that this shouldn't just apply to `Target` - a core tenet of good UX is a series of well understood patterns that the user can re-apply through-out the use of a product.

> move the argparse code example to Guide-level explanation. i think the example `--help` output could stay as motivation.

I've duplicated it a bit to ensure it makes sense in the Guide-level explanation.